infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -182,23 +182,18 @@ public ServerContext(Map<String, Object> serverConfig) {
     }
 
     int getBaseServerThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--base-server-threads");
-        return threadCount == null ? 1 : threadCount;
-    }
-
-    int getLayoutServerThreadCount() {
-        Integer threadCount = getServerConfig(Integer.class, "--layout-server-threads");
-        return threadCount == null ? 1 : threadCount;
+        int threadCount = Integer.parseInt(getServerConfig(String.class, "--base-server-threads"));
+        return threadCount == 0 ? 1 : threadCount;
```

Comment:
Just out of curiosity, how does change help fix the related bugs/issues?

Reply:
On master if you pass any parameters that change the number of server threads It will fail because the values in the opts map will always be String, and its being casted to an Integer without proper conversion. 

You can see the issue if you run master with `./bin/corfu_server -ms --logunit-threads 4 9000`

Reply:
Gotcha, thanks!